tests/performance_tests/performance_test_basic.py

```diff
@@ -480,28 +509,87 @@ def captureLowLevelArtifacts(self):
+        report['Analysis']['ForkedBlocks'] = {}
+        report['Analysis']['ForksCount'] = {}
+        report['Analysis']['DroppedBlocks'] = {}
+        report['Analysis']['DroppedBlocksCount'] = {}
+        for nodeNum in range(0, self.data.numNodes):
+            formattedNodeNum = str(nodeNum).zfill(2)
+            report['Analysis']['ForkedBlocks'][formattedNodeNum] = self.data.forkedBlocks[formattedNodeNum]
+            report['Analysis']['ForksCount'][formattedNodeNum] = len(self.data.forkedBlocks[formattedNodeNum])
+            report['Analysis']['DroppedBlocks'][formattedNodeNum] = self.data.droppedBlocks[formattedNodeNum]
+            report['Analysis']['DroppedBlocksCount'][formattedNodeNum] = len(self.data.droppedBlocks[formattedNodeNum])
+        report['args'] =  argsDict
+        report['env'] = {'system': system(), 'os': os.name, 'release': release(), 'logical_cpu_count': os.cpu_count()}
+        report['nodeosVersion'] = nodeosVers
+        return report
 
     def analyzeResultsAndReport(self, testResult: PtbTpsTestResult):
         args = self.prepArgs()
         artifactsLocate = log_reader.ArtifactPaths(nodeosLogDir=self.nodeosLogDir, nodeosLogPath=self.nodeosLogPath, trxGenLogDirPath=self.trxGenLogDirPath, blockTrxDataPath=self.blockTrxDataPath,
                                                    blockDataPath=self.blockDataPath, transactionMetricsDataPath=self.transactionMetricsDataPath)
         tpsTestConfig = log_reader.TpsTestConfig(targetTps=self.ptbConfig.targetTps, testDurationSec=self.ptbConfig.testTrxGenDurationSec, tpsLimitPerGenerator=self.ptbConfig.tpsLimitPerGenerator,
-                                                 numBlocksToPrune=self.ptbConfig.numAddlBlocksToPrune, numTrxGensUsed=testResult.numGeneratorsUsed,
-                                                 targetTpsPerGenList=testResult.targetTpsPerGenList, quiet=self.ptbConfig.quiet)
-        self.report = log_reader.calcAndReport(data=self.data, tpsTestConfig=tpsTestConfig, artifacts=artifactsLocate, argsDict=args, testStart=self.testStart,
-                                               completedRun=testResult.completedRun, nodeosVers=self.clusterConfig.nodeosVers, targetApiEndpoint=self.ptbConfig.endpointApi)
+                                                 numBlocksToPrune=self.ptbConfig.numAddlBlocksToPrune, numTrxGensUsed=testResult.numGeneratorsUsed, targetTpsPerGenList=testResult.targetTpsPerGenList,
+                                                 quiet=self.ptbConfig.quiet, printMissingTransactions=self.ptbConfig.printMissingTransactions)
+        self.logAnalysis = log_reader.analyzeLogResults(data=self.data, tpsTestConfig=tpsTestConfig, artifacts=artifactsLocate)
+        self.testEnd = datetime.utcnow()
+
+        self.testResult = PerformanceTestBasic.PerfTestBasicResult(targetTPS=self.ptbConfig.targetTps, resultAvgTps=self.logAnalysis.tpsStats.avg, expectedTxns=self.ptbConfig.expectedTransactionsSent,
```

Comment:
I commend to your attention the `**` operator.  Briefly, given a function with signature:
```
def func(a, b, c):
    print(a, b, c)
```
and a dictionary with corresponding members:
```
args = {'a': 1, 'b': 2, 'c': 3}
```
`func can be invoked as:
```
func(**args)
```
Notably, this works with `dataclasses.asdict`.  It converts all of the member names to strings so the `**` operator works.  The function signature is very flexible as well, since Python does not require any particular order of keyword arguments.  The only requirement is that every keyword passed in must be accepted by the function.  Conversely, not all keywords the function accepts must be in the dictionary and the operator can be used multiple times within the same function call, in any order, and interspersed with explicit keyword parameters, e.g.
```
funcComplex(somearg='argument', **firstArgs, somekeyword=1, **moreArgs)
```

Reply:
Will address this in follow on work.